packages/form-js-viewer/src/render/components/form-fields/Taglist.js

```diff
@@ -50,43 +49,43 @@ export default function Taglist(props) {
     values: options
   } = useValuesAsync(field);
 
+  // We cache a map of option values to their index so that we don't need to search the whole options array every time to correlate the label
+  const valueToOptionMap = useMemo(() => Object.assign({}, ...options.map((o, x) => ({ [o.value]:  options[x] }))), [ options ]);
 
   // Usage of stringify is necessary here because we want this effect to only trigger when there is a value change to the array
   useEffect(() => {
     if (loadState === LOAD_STATES.LOADED) {
-      const selectedValues = values.map(v => options.find(o => o.value === v)).filter(v => v !== undefined);
-      setSelectedValues(selectedValues);
+      setFilteredOptions(options.filter((o) => o.label && (o.label.toLowerCase().includes(filter.toLowerCase())) && !values.includes(o.value)));
     }
     else {
-      setSelectedValues([]);
+      setFilteredOptions([]);
     }
-  }, [ JSON.stringify(values), options, loadState ]);
+  }, [ filter, JSON.stringify(values), options, loadState ]);
 
   useEffect(() => {
-    if (loadState === LOAD_STATES.LOADED) {
-      setFilteredValues(options.filter((o) => o.label && (o.label.toLowerCase().includes(filter.toLowerCase())) && !values.includes(o.value)));
-    }
-    else {
-      setFilteredValues([]);
-    }
-  }, [ filter, JSON.stringify(values), options ]);
-
-  useEffect(() => {
-    setHasValuesLeft(selectedValues.length < options.length);
-  }, [ selectedValues.length, options.length ]);
+    setHasOptionsLeft(options.length > values.length);
+  }, [ options.length, values.length ]);
 
   const onFilterChange = ({ target }) => {
     setIsEscapeClose(false);
     setFilter(target.value);
   };
 
-  const selectValue = (option) => {
-    setFilter('');
-    props.onChange({ value: [ ...values, option.value ], field });
+  const selectValue = (value) => {
+    if (filter) {
+      setFilter('');
+    }
+
+    // Ensure values cannot be double selected due to latency
+    if (values.at(-1) === value) {
```

Comment:
I think we don't guarantee support for IE anymore, so this might be fine.

https://caniuse.com/?search=Array%3A%20at

Just to make sure, is there maybe a safer choice instead of using `.at` ?

Reply:
Yeah I've switched over to the good ol' a[a.length-1]